protocol/requests/enable_installation_and_pair.go

```diff
@@ -8,6 +8,9 @@ var ErrEnableInstallationAndPairInvalidID = errors.New("enable installation and
 
 type EnableInstallationAndPair struct {
 	InstallationID string `json:"installationId"`
+
+	// used for testing
+	getInstallationID func() string
```

Comment:
Although I like the simplicity, I don't think it's a good approach. Production code should not contain test things.

I think it should be feasible to unit-test `saveDataAndPrepareResponse` directly, by passing request with specific `installationID`, can you please try?

Reply:
If you find it not possible to unit-test, then we should go the proper mock way.
1. Keep `EnableInstallationAndPair` as before, no custom callback properties
2. Define `EnableInstallationAndPairInterface` interface with `GetInstallationID` method
3. Implement `func (r *EnableInstallationAndPair) GetInstallationID()`
4. Change `EnableInstallationAndPair` signature to take `EnableInstallationAndPairInterface`.
5. Use `gomock` to generate a mock in tests
6. Use the generated mocks in test

Reply:
> _`saveDataAndPrepareResponse` is a bit complicated and composed_

This is in fact a very good reason for unit-testing 🙂

Reply:
Well I didn't say it's gonna be easier 😄 
But in the future we should try hard to write more unit tests than integration tests.